shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs

```diff
@@ -1459,6 +1469,111 @@ fn test_source_tls_and_single_tls() {
     test_batch_statements(&connection);
 }
 
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_events_keyspace() {
+    let mut rng = rand::thread_rng();
+
+    let _docker_compose = DockerCompose::new(
+        "tests/test-configs/cassandra-peers-rewrite/docker-compose-4.0-cassandra.yaml",
+    );
+
+    let _shotover_manager = ShotoverManager::from_topology_file(
+        "tests/test-configs/cassandra-peers-rewrite/topology.yaml",
+    );
+
+    let user = "cassandra";
+    let password = "cassandra";
+    let auth = StaticPasswordAuthenticatorProvider::new(&user, &password);
+    let config = NodeTcpConfigBuilder::new()
+        .with_contact_point("127.0.0.1:9044".into())
+        .with_authenticator_provider(Arc::new(auth))
+        .build()
+        .await
+        .unwrap();
+
+    let session = TcpSessionBuilder::new(RoundRobinLoadBalancingStrategy::new(), config).build();
+    let mut event_recv = session.create_event_receiver();
+
+    let mut tries = 0;
+    loop {
+        if tries > 3 {
```

Comment:
Why do we have a retries system?
Shouldnt the timeout on event_recv.recv() be enough?

Reply:
I need to confirm this with someone that has knowledge of Cassandra internals, but I tested it with just a straight connection to Cassandra with the driver. It's possible for Cassandra to lose events and not deliver them to clients that are subscribed, so I've allowed 3 tries.

Reply:
I see, that research would be much appreciated.
I would want to see some kind of documentation or demonstration of this behavior before allowing this retry.
We dont want our tests to allow flakiness to be introduced into shotover

Reply:
The issue was that the Cassandra driver was not finished registering for the event before the keyspace was created and the event fired off.

The issue ideally should be fixed within the driver itself. It should not return a `Session` object that is technically not ready to use yet. I can add a sleep to the test for the time being.